dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/security/LibertyJaxRsClientSSLOutInterceptor.java

```diff
@@ -127,26 +130,52 @@ private TLSClientParameters retriveHTTPTLSClientParametersUsingSSLRef(HTTPCondui
             //let's use liberty SSL configuration
             tlsClientParams.setSSLSocketFactory(sslSocketFactory);
             tlsClientParams.setDisableCNCheck(disableCNCheck);
+            tlsClientParams.setCipherSuites(getSSLCipherSuites(sslRef, message));
+            // TODO: this is something we may want to do in the future, however, CXF currently
+            // only allows us to set 1 protocol
+            // tlsClientParams.setSecureSocketProtocol(getSSLProtocols(sslRef, message));
         } else if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(tc, "May not enable feature ssl-1.0 or appSecurity-2.0.");
         }
 
+        // Get the ciphers
+
+        // Get the protocols
+
         return tlsClientParams;
     }
 
     public void setTLSConfiguration(TLSConfiguration secConfig) {
         this.secConfig = secConfig;
     }
 
-    private SSLSocketFactory getSocketFactory(String sslRef, Message message) {
-        try {
-            final Class<?> jaxrsSslMgrClass = Class.forName("com.ibm.ws.jaxrs20.appsecurity.security.JaxRsSSLManager");
-            if (jaxrsSslMgrClass == null) {
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "getSocketFactory could not find JaxRsSSLManager class");
-                }
-                return null;
+    private Class<?> getJaxRsSSLManagerClass() throws ClassNotFoundException {
+        final Class<?> jaxrsSslMgrClass = Class.forName("com.ibm.ws.jaxrs20.appsecurity.security.JaxRsSSLManager");
+        if (jaxrsSslMgrClass == null) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getSocketFactory could not find JaxRsSSLManager class");
             }
+        }
+        return jaxrsSslMgrClass;
+    }
+
+    private Object[] getJaxRsSSLManagerParams(String sslRef, Message message) {
+        String uriString = (String) message.get(Message.REQUEST_URI);
+        URI uri = URI.create(uriString);
+        int port = uri.getPort();
+
+        // if the port wasn't specified, use the default SSL port (443)
+        if (port == -1 && !uriString.contains(":-1")) {
+            port = 443;
```

Comment:
How about putting a tr.debug statement here saying that the port wasn't specified.